tests/test_extract_imports_errors.py

```diff
@@ -76,6 +79,13 @@ def test_parse_code__on_parse_error__logs_error(caplog):
     assert f"Could not parse code from {source}" in caplog.text
 
 
+def test_parse_any_args__no_stdin_provided__logs_error():
+    stdin = None
+    arg = "<stdin>"
+    with pytest.raises(StdInNotProvidedException):
+        parse_any_arg(arg, stdin)
```

Comment:
Related to my reservations about `StdInNotProvidedException`, I'm questioning the reason for this test.

I think it tests something that a CLI user should not be able to trigger in any way, and I think if we make `parse_any_arg()` take a `stdin: TextIO = io.StringIO()` argument instead, we can avoid `StdInNotProvidedException` altogether. What do you think?

Reply:
Agreed.